### ISSUE TITLE
history merge test fails on OpenBSD

### ISSUE THREAD

**Issue by faho** [fish-shell/fish-shell#6477]
Again, on sr.ht (https://builds.sr.ht/~faho/job/133042), this time with OpenBSD:

```

Testing history merge
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Error: Test failed on line 3889: history_contains(reader, more_texts[j])
Testing history race conditions
Error: Item dropped from history: 90318 113
Error: Item dropped from history: 65161 38
Error: Item dropped from history: 65161 37
Error: Item dropped from history: 65161 36
Error: Item dropped from history: 65161 35
Error: Item dropped from history: 65161 34
Error: Item dropped from history: 65161 33
Error: Item dropped from history: 65161 32
Error: Item dropped from history: 65161 31
Error: Item dropped from history: 65161 30
Error: Item dropped from history: 65161 29
Error: Item dropped from history: 65161 28
```

and so on.

(again with 33a234e19, which is 484e590b7 without a rebase)

**Comment by faho** [comment]
Supposed to be fixed by 3ed8a57bc59c931e890c60df38222f4532d09b66.

**Comment by faho** [comment]
Yeah, seems fixed.
